### PR TITLE
[global] Implement simple global pull

### PR DESCRIPTION
**Pull request by mikeland86**
## Summary

This "pulls" a global config either from a local path or URL. It merges the packages into the existing config.

## How was it tested?

```
devbox global pull devbox.json
devbox global pull https://raw.githubusercontent.com/jetpack-io/devbox/main/devbox.json
```

### CODE REVIEW

File: internal/boxcli/global.go [hunk 1]
```diff
@@ -86,6 +87,16 @@ func globalListCmd() *cobra.Command {
 	}
 }
 
+func globalPullCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:     "pull <file> | <url>",
+		Short:   "Pulls a global config from a file or URL",
```
Comment (savil) [inline internal/boxcli/global.go:93]: it seems that url requires the scheme. That's non-obvious but very reasonable, so could we specify an example in the Long explanation?

File: internal/impl/global.go [hunk 2]
```diff
@@ -70,21 +70,57 @@ func (d *Devbox) RemoveGlobal(pkgs ...string) error {
 	return d.saveCfg()
 }
 
+func (d *Devbox) PullGlobal(path string) error {
+	u, err := url.Parse(path)
+	if err == nil && u.Scheme != "" {
```
Comment (savil) [inline internal/impl/global.go:75]: scheme requirement is here (ref. previous comment above)

File: internal/impl/global.go [hunk 3]
```diff
@@ -95,6 +131,16 @@ func GlobalConfigPath() (string, error) {
 	return filepath.Join(home, "/.config/devbox/"), nil
 }
 
+func globalProfilePath() (string, error) {
+	configPath, err := GlobalConfigPath()
+	if err != nil {
+		return "", err
+	}
+	nixDirPath := filepath.Join(configPath, "nix")
+	_ = os.MkdirAll(nixDirPath, 0755)
```
Comment (savil) [inline internal/impl/global.go:140]: lets check the err

**Review by savil — APPROVED** [review]

**Comment by mikeland86** [comment]
@savil to avoid rebase conflicts, I addressed your comments in https://github.com/jetpack-io/devbox/pull/637

I'll merge both at same time.

**Pull request merged**